src/librustc_resolve/macros.rs

```diff
@@ -156,16 +156,20 @@ impl<'a> base::Resolver for Resolver<'a> {
         self.whitelisted_legacy_custom_derives.contains(&name)
     }
 
-    fn visit_expansion(&mut self, mark: Mark, expansion: &Expansion, derives: &[Mark]) {
+    fn visit_expansion(&mut self, mark: Mark, expansion: &Expansion, derives: &[(Mark, Path)]) {
```

Comment:
I would think about making this new tuple into a struct - it's OK for now, but I think it is a bit opaque to identify a derive by a `(Mark, Path)`.